pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -383,6 +407,18 @@ func (r *hibernationReconciler) canHibernate(cd *hivev1.ClusterDeployment) (bool
 	return true, "Hibernation capable"
 }
 
+func (r *hibernationReconciler) syncSetsApplied(cd *hivev1.ClusterDeployment) (bool, string) {
+	clusterSync := &hiveintv1alpha1.ClusterSync{}
+	err := r.Get(context.Background(), types.NamespacedName{Namespace: cd.Namespace, Name: cd.Name}, clusterSync)
+	if err != nil {
+		return false, fmt.Sprintf("could not get ClusterSync: %v", err)
```

Comment:
feels like we may need to return an error here and force a requeue rather than following the path for normal execution when we're waiting for syncsets to reach first success.

Reply:
Thinking about this I'm not sure that we need to error and go into back off if we're requeuing for the time to force hibernate. I could do this though by squashing the returns from setting the condition and returning an err from reconcile.

Reply:
Suppose it would hibernate sooner in the event that the syncsets did succeed.